src/Components/Components/src/BindConverter.cs

```diff
@@ -1933,6 +1962,14 @@ internal static class ParserDelegateCache
                 {
                     parser = ConvertToNullableTimeOnly;
                 }
+                else if (typeof(T) == typeof(Guid))
+                {
+                    parser = ConvertToGuid;
+                }
+                else if (typeof(T) == typeof(Guid?))
```

Comment:
Kinda unrelated - I imagine this block of code would look a lot nicer as a switch expression.

Reply:
Yes, probably. This code predates switch expressions. Keeping this out-of-scope for the PR though.